diff -ruN a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,20 +1,11 @@
 AUTOMAKE_OPTIONS = foreign
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = libcnary src include tools test docs
-
-if HAVE_CYTHON
-SUBDIRS += cython
-endif
-
-if BUILD_FUZZERS
-SUBDIRS += fuzz
-endif
+SUBDIRS = libcnary src include
 
 EXTRA_DIST = \
 	README.md
 
 docs/html: $(top_builddir)/doxygen.cfg $(top_srcdir)/include/plist/*.h
-	rm -rf docs
 	doxygen doxygen.cfg
 
-docs: doxygen.cfg docs/html
+docs: doxygen.cfg
diff -ruN a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -107,38 +107,7 @@
 	AC_DEFINE(HAVE_TM_TM_ZONE, 1, [Define if struct tm has a tm_zone member])
 fi
 
-# Cython Python Bindings
-AC_ARG_WITH([cython],
-            [AS_HELP_STRING([--without-cython],
-            [build Python bindings using Cython (default is yes)])],
-            [build_cython=false],
-            [build_cython=true])
-if test "$build_cython" = "true"; then
-            AM_PATH_PYTHON(2.3)
-            AC_PROG_CYTHON(0.17.0)
-            CYTHON_PYTHON
-else
-            CYTHON=false
-fi
-if [test "x$CYTHON" != "xfalse"]; then
-            PKG_PROG_PKG_CONFIG
-            AC_MSG_CHECKING([for libplist Cython bindings])
-            CYTHON_PLIST_INCLUDE_DIR=$($PKG_CONFIG --variable=includedir libplist-2.0)/plist/cython
-            if [test ! -d "$CYTHON_PLIST_INCLUDE_DIR"]; then
-                CYTHON_PLIST_INCLUDE_DIR=.
-                cython_python_bindings=yes
-                AC_MSG_RESULT([Using built-in libplist Cython bindings (assuming this is a first build)])
-            else
-                AC_SUBST([CYTHON_PLIST_INCLUDE_DIR])
-                AC_MSG_RESULT([$CYTHON_PLIST_INCLUDE_DIR])
-                cython_python_bindings=yes
-            fi
-else
-            cython_python_bindings=no
-fi
-AM_CONDITIONAL([HAVE_CYTHON],[test "x$cython_python_bindings" = "xyes"])
-
-AS_COMPILER_FLAGS(GLOBAL_CFLAGS, "-Wall -Wextra -Wredundant-decls -Wshadow -Wpointer-arith  -Wwrite-strings -Wswitch-default -Wno-unused-parameter -Wno-strict-aliasing -fvisibility=hidden $PTHREAD_CFLAGS")
+AS_COMPILER_FLAGS(GLOBAL_CFLAGS, "-Wall -Wextra -Wredundant-decls -Wshadow -fPIC -Wpointer-arith -Wwrite-strings -Wswitch-default -Wno-unused-parameter -Wno-strict-aliasing -fvisibility=hidden $PTHREAD_CFLAGS")
 GLOBAL_LDFLAGS="$PTHREAD_LIBS"
 
 AC_ARG_ENABLE(debug,
@@ -164,104 +133,6 @@
         AC_DEFINE([HAVE_FVISIBILITY], [1], [Define if compiled with -fvisibility=hidden])
 esac
 
-AC_ARG_WITH([sanitizers],
-            [AS_HELP_STRING([--with-sanitizers],
-            [build libplist with sanitizers (default is no)])],
-            [build_sanitizers=${withval}],
-            [])
-
-AC_ARG_WITH([fuzzers],
-            [AS_HELP_STRING([--with-fuzzers],
-            [build fuzzers (implies --with-sanitizers, default is no)])],
-            [build_fuzzers=${withval}],
-            [build_fuzzers=no])
-
-if test "x$build_fuzzers" = "xyes"; then
-    if test "x$build_sanitizers" = "xno"; then
-        AC_MSG_ERROR([--with-fuzzers implies --with-sanitizers, but --without-sanitizers was given. This does not work.])
-    fi
-    build_sanitizers=yes
-fi
-
-if test "x$build_sanitizers" = "xyes"; then
-    AS_COMPILER_FLAG([-fsanitize=address], [
-        SANITIZER_FLAGS+=" -fsanitize=address"
-        ASAN_AVAILABLE=yes
-    ], [])
-    if test "$ASAN_AVAILABLE" = "yes"; then
-        AS_COMPILER_FLAG([-fsanitize=address -fsanitize-address-use-after-scope], [
-            SANITIZER_FLAGS+=" -fsanitize-address-use-after-scope"
-        ], [])
-        SANITIZERS+="ASAN "
-    fi
-
-    AS_COMPILER_FLAG([-fsanitize=undefined], [
-        SANITIZER_FLAGS+=" -fsanitize=undefined"
-        UBSAN_AVAILABLE=yes
-    ], [])
-
-    if test "$UBSAN_AVAILABLE" = "yes"; then
-        SANITIZERS+="UBSAN "
-    fi
-
-    if test -z "$SANITIZER_FLAGS"; then
-        AC_MSG_ERROR([compiler doesn't support -fsanitize=address nor -fsanitize=undefined])
-    fi
-
-    COVERAGE_CHECKS="trace-pc-guard trace-cmp edge"
-    for COV_CHECK in $COVERAGE_CHECKS; do
-        AS_COMPILER_FLAG([-fsanitize-coverage=$COV_CHECK], [
-            if test -z "$SAN_COV_FLAGS"; then
-                SAN_COV_FLAGS="$COV_CHECK"
-            else
-                SAN_COV_FLAGS+=",$COV_CHECK"
-            fi
-        ], [])
-    done
-    if test -n "$SAN_COV_FLAGS"; then
-        SANITIZER_FLAGS+=" -fsanitize-coverage=$SAN_COV_FLAGS"
-        SANITIZERS+="+coverage "
-    else
-        AC_MSG_WARN([No sanitizer coverage supported by compiler])
-    fi
-
-    CFLAGS="-O1"
-
-    AS_COMPILER_FLAG([-fno-omit-frame-pointer], [
-        CFLAGS+=" -fno-omit-frame-pointer"
-    ], [])
-
-    AS_COMPILER_FLAG([-gline-tables-only], [
-        CFLAGS+=" -gline-tables-only"
-    ],
-    [
-        CFLAGS+=" -g"
-    ])
-
-    EXTRA_CONF+="  Enabled sanitizers ......: $SANITIZERS
-"
-
-    CFLAGS+=" $SANITIZER_FLAGS"
-    CXXFLAGS="$CFLAGS -std=c++11"
-fi
-
-if test "x$build_fuzzers" = "xyes"; then
-    if test "$CXX" != "clang++"; then
-        AC_MSG_WARN([building fuzzers requires clang/clang++ (continuing anyway)])
-    fi
-
-    CFLAGS+=" -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION"
-
-    EXTRA_CONF+="  Build fuzzers ...........: yes
-"
-fi
-
-AM_CONDITIONAL([BUILD_FUZZERS],[test "x$build_fuzzers" = "xyes"])
-
-if test "x$build_fuzzers" = "xyes" || test "x$build_sanitizers" = "xyes"; then
-    AS_COMPILER_FLAGS(TEST_CFLAGS, [$CFLAGS])
-fi
-
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 
 AC_OUTPUT([
@@ -269,13 +140,7 @@
 libcnary/Makefile
 src/Makefile
 src/libplist-2.0.pc
-src/libplist++-2.0.pc
 include/Makefile
-tools/Makefile
-docs/Makefile
-cython/Makefile
-test/Makefile
-fuzz/Makefile
 doxygen.cfg
 ])
 
@@ -285,7 +150,6 @@
 
   Install prefix ..........: $prefix
   Debug code ..............: $debug
-  Python bindings .........: $cython_python_bindings
 $EXTRA_CONF
   Now type 'make' to build $PACKAGE $VERSION,
   and then 'make install' for installation.
diff -ruN a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -7,8 +7,7 @@
 AM_LDFLAGS = $(GLOBAL_LDFLAGS)
 
 lib_LTLIBRARIES = \
-	libplist-2.0.la \
-	libplist++-2.0.la
+	libplist-2.0.la
 
 libplist_2_0_la_LIBADD = $(top_builddir)/libcnary/libcnary.la
 libplist_2_0_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(LIBPLIST_SO_VERSION) -no-undefined
@@ -59,5 +58,4 @@
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = \
-	libplist-2.0.pc \
-	libplist++-2.0.pc
+	libplist-2.0.pc
